commands/new_function.go

```diff
@@ -99,6 +102,24 @@ func runNewFunction(cmd *cobra.Command, args []string) error {
 		if _, statErr := os.Stat(appendFile); statErr != nil {
 			return fmt.Errorf("unable to find file: %s - %s", appendFile, statErr.Error())
 		}
+
+		originalBytes, readErr = ioutil.ReadFile(appendFile)
+		if readErr != nil {
+			fmt.Printf("unable to read %s to append, %s", appendFile, readErr)
+		}
+
+		services, parseErr := stack.ParseYAMLData(originalBytes, "", "")
+
+		if parseErr != nil {
+			fmt.Printf("Error parsing %s yml file", appendFile)
+		}
+
+		for function := range services.Functions {
```

Comment:
`services.Functions` is a map object. You don't need to loop through it to search for the function:

```		if _, ok := services.Functions[functionName]; ok {
			return fmt.Errorf(`Function %s already exists in %s file. 
 Cannot have duplicate function names in same yml file`, functionName, appendFile)
		}```